### PR TITLE
Issue 1722/sub org survey read only

### LINKED ISSUE #1722: Accessing an inherited survey presents as if sub-org admin can edit it
## Description
When a survey is inherited to a sub-organization, admins of that sub-organization should only be allowed to see the submissions, not edit the survey in any way. But navigating to such a survey today (and even more relevantly, after a fix to #1721) leads to a normal survey page where things seem like they're editable, but editing them as a sub-org admin will (correctly) be prevented by the API and raise errors.

## Steps to reproduce
1. Go to http://localhost:3000/organize/1/projects (parent org)
2. Click on any survey (go via "Activities" if none shows up in the overview)
3. Go to the "submissions" tab
4. Make sure "share with sub-organizations" is activated
5. Change the URL to replace the org ID 1 with 2 (just clicking won't work at the time of writing because of #1721)
6. Ono the survey page that loads, interact with the title

### Expected Behaviour
The title should be read-only, because you are acting within an organization that does not own this survey.

### Actual Behaviour
The title can be edited. If you had been logged in as a sub-org admin, editing it would also throw an error, but that's secondary, because the interface should not present as editable in the first place.

## Screenshots (if you have any)
image

## Proposed solutions
When visiting an inherited survey, disable all editable fields. Present a banner somewhere saying something along the lines of "This survey is owned by ORG TITLE so you can't edit it" if the user is not an admin in the parent org, or "This survey is owned by ORG_TITLE. Go to LINK to edit it" if the user is an admin in the parent org.

Editable fields that need to be disabled include:

- [x] The title
- [x] The "publish" button and scheduling widget
- [x] The content in the "questions" section
- [x] Under submissions, the "share with sub-organizations" option

### PR DESCRIPTION
**Pull request by kaulfield23**
## Description
This PR fixes a bug where sub-organizations were able to edit the original survey shared by the parent organization. It also implements a 'shared with us' project in overview which makes it easier for users to find shared activities.

## Screenshots
image
image
image

### Shared with us
image
image
image
image

## Changes

* Adds messageIds to relevant places
* Changes `.tsx` to `.ts` of hooks in organization dir
* Adds `isReadOnly` to `ZUIDateRangePicker` and `ZUIEditTextInPlace`
* Adds `<Alert/>` to `TabbedLayout`
* Adds `shared/index.tsx`, `shared/acitivities.tsx` and `shared/archive.tsx`
* Creates `SharedCard.tsx`

## Related issues
Resolves #1722

### CODE REVIEW

**Review by richardolsson — COMMENTED** [review]
Nice work! I began reviewing this but didn't finish so I didn't publish my comment. I'm publishing them now and will continue to review the rest of the PR as soon as I can, but most likely not until next week.

File: src/pages/organize/[orgId]/projects/index.tsx [hunk 1]
```diff
@@ -55,10 +58,18 @@ const AllCampaignsSummaryPage: PageWithLayout = () => {
...
 
   if (onServer) {
     return null;
   }
+  //The shared card is currently only visible when there are shared surveys, but there will be more shared activities in the future.
+  const sharedSurveys = surveys.filter(
+    (survey) =>
+      survey.org_access === 'suborgs' &&
+      survey.organization.id === orgId &&
+      !dayjs(survey.expires).isBefore(dayjs(), 'day')
```
[the remark this answers is not in the document]
Comment (kaulfield23) [reply src/pages/organize/[orgId]/projects/index.tsx:71]: In Figma there is a note about shared project that says 'Visible only when shared activities are available'. I interpreted this to mean that the 'Share with us' will only be visible when there are ongoing activities. I'm going to check with Josef!

**Comment by kaulfield23** [comment]
Come to think of it, I think <SharedCard/> should not be in survey dir. I will fix it today!

File: src/features/surveys/l10n/messageIds.ts [hunk 2]
```diff
@@ -9,6 +9,15 @@ export default makeMessages('feat.surveys', {
     textButton: m('Text'),
     title: m('Choose a block type to add more content to your survey'),
   },
+  alert: {
+    editable: m<{ orgTitle: string }>(
+      'This survey is owned by {orgTitle} and is editable only from its original location.'
+    ),
+    goOriginal: m('Go to original'),
+    notEditable: m<{ orgTitle: string }>(
+      'This survey is owned by {orgTitle} and it is not editable.'
```
Comment (richardolsson) [inline src/features/surveys/l10n/messageIds.ts:18]: Maybe a more direct text would be:

```suggestion
      'This survey is owned by {orgTitle} and you can not edit it.'
```

File: src/features/surveys/l10n/messageIds.ts [hunk 3]
```diff
@@ -9,6 +9,15 @@ export default makeMessages('feat.surveys', {
     textButton: m('Text'),
     title: m('Choose a block type to add more content to your survey'),
   },
+  alert: {
+    editable: m<{ orgTitle: string }>(
+      'This survey is owned by {orgTitle} and is editable only from its original location.'
```
Comment (richardolsson) [inline src/features/surveys/l10n/messageIds.ts:14]:
```suggestion
      'This survey is owned by {orgTitle} and you can only edit it from its original location.'
```

File: src/features/campaigns/components/SharedCard.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          </Typography>
+          <Box sx={{ display: 'flex', mr: '0.3rem', position: 'relative' }}>
+            <GroupWork
+              color="secondary"
+              sx={{ backgroundColor: 'white', borderRadius: '50%', zIndex: 1 }}
+            />
+            <GroupWork
+              color="secondary"
+              sx={{ left: 8, position: 'absolute' }}
+            />
```
Comment (richardolsson) [inline src/features/campaigns/components/SharedCard.tsx:38]: Nice solution!

File: src/features/campaigns/l10n/messageIds.ts [hunk 5]
```diff
@@ -142,6 +142,30 @@ export default makeMessages('feat.campaigns', {
...
+      'All ongoing activities shared with your organization will appear here.'
+    ),
+    noArchives: m(
+      'All expired activities that have been shared with your organization will appear here.'
+    ),
+    title: m('Shared with us'),
+  },
+  sharedLayout: {
+    alertMsg: m(
+      'This project contains only activities shared from other organizations and it is viewable only.'
```
Comment (richardolsson) [inline src/features/campaigns/l10n/messageIds.ts:157]:
```suggestion
      'This project contains only activities shared from other organizations and you can view it but not change it.'
```

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work on this! I think it's really nice. I've looked at the code and played around with the preview build. There were some texts that I feel could be improved when I saw them in place, but then I think it's ready to merge!

**Review by richardolsson — DISMISSED** [review]
With those minor changes to the messages I think this is ready to merge now. Nice work! 🎉

**Comment by kaulfield23** [comment]
Just fixed a merge conflict!

**Review by richardolsson — APPROVED** [review]
Full speed ahead! 🚀

**Pull request merged**